drizzlepac/generate_final_product_filenames.py

```diff
@@ -1,15 +1,171 @@
+    """
+    Interpret the database query for a given obset to prepare the returned
+    values for use in generating the names of all the expected output products.
+
+    Input will have format of:
+        ia1s70jtq_flt.fits,11150,A1S,70,70,149.232269,F110W,IR,/ifs/archive/ops/hst/public/ia1s/ia1s70jtq/ia1s70jtq_flt.fits
+        ia1s70iwq_flt.fits,11150,A1S,70,70,0.91161000000000003,F160W,IR,/ifs/archive/ops/hst/public/ia1s/ia1s70iwq/ia1s70iwq_flt.fits
+        which are
+        filename, proposal_id, program_id, obset_id, visit_id, exptime, filters, detector, pathname
+
+    Output dict will have format (as needed by further code for creating the
+        product filenames) of:
+
+        obs_info_dict["single exposure product 00"] = {'info': '11150 A1S WFC3 IR F110W IA1S70JTQ',
+                                                        'files':['ia1s70jtq_flt.fits']}
+        obs_info_dict["single exposure product 01"] = {'info': '11150 A1S WFC3 IR F160W IA1S70JWQ',
+                                                        'files':['ia1s70jwq_flt.fits']}
+
+        obs_info_dict["filter product 00"] = {"info": '11150 A1S WFC3 IR F110W',
+                                              "files":['ia1s70jtq_flt.fits']}
+        obs_info_dict["filter product 01"] = {"info": '11150 A1S WFC3 IR F160W',
+                                              "files":['ia1s70jwq_flt.fits']}
+        obs_info_dict["total detection product 00"] = {'info': '11150 A1S WFC3 IR',
+                                                       'files':['ia1s70jtq_flt.fits',
+                                                                'ia1s70iwq_flt.fits']}
+
+    """
+    colnames = ['filename', 'proposal_id', 'program_id', 'obset_id', 'visit_id',
+                'exptime', 'filters', 'detector', 'pathname']
+    obset_table = Table.read(results, format='ascii.fast_no_header', names=colnames)
+    # Add INSTRUMENT column
+    instr = INSTRUMENT_DICT[obset_table['filename'][0][0]]
+    # convert input to an Astropy Table for parsing
+    obset_table.add_column(Column([instr] * len(obset_table)), name='instrument')
+    # parse Table into a tree-like dict
+    obset_tree = build_obset_tree(obset_table)
+    # Now create final dict
+    obset_dict = parse_obset_tree(obset_tree)
+
+    return obset_dict
+
+# Translate the database query on an obset into actionable lists of filenames
+def build_obset_tree(obset_table):
+    """Convert obset table into a tree listing all products to be created."""
+
+    # Each product will consist of the appropriate string as the key and
+    # a dict of 'info' and 'files' information
+
+    # Start interpreting the obset table
+    obset_tree = {}
+    for row in obset_table:
+        # Get some basic information from the first row - no need to check
+        # for multiple instruments as data from different instruments will
+        # not be combined.
+        det = row['detector']
+        filt = row['filters']
```

Comment:
Something needs to be done with the value of `filt` here to remove extraneous 'CLEAR' entries and to convert the ';' to '-' as well, unless this conversion is expected to take place later in the code when the final product names are generated.